src/wallet/gtest/test_rpc_wallet.cpp

```diff
@@ -0,0 +1,355 @@
+    EXPECT_EQ(0, chainActive.Height());
+    wtx.SetMerkleBranch(block);
+    pwalletMain->LoadWalletTx(wtx);
+
+    // Context that z_sendmany requires
+    auto builder = TransactionBuilder(consensusParams, nextBlockHeight, std::nullopt, pwalletMain);
+    mtx = CreateNewContextualCMutableTransaction(consensusParams, nextBlockHeight);
+
+    auto selector = pwalletMain->ZTXOSelectorForAddress(taddr, true, false).value();
+    std::vector<SendManyRecipient> recipients = { SendManyRecipient(std::nullopt, pa, 1*COIN, "ABCD") };
+    TransactionStrategy strategy(PrivacyPolicy::AllowRevealedSenders);
+    std::shared_ptr<AsyncRPCOperation> operation(new AsyncRPCOperation_sendmany(std::move(builder), selector, recipients, 0, strategy));
+    std::shared_ptr<AsyncRPCOperation_sendmany> ptr = std::dynamic_pointer_cast<AsyncRPCOperation_sendmany> (operation);
+
+    // Enable test mode so tx is not sent
+    static_cast<AsyncRPCOperation_sendmany *>(operation.get())->testmode = true;
+
+    // Generate the Sapling shielding transaction
+    operation->main();
+    if (!operation->isSuccess()) {
+        FAIL() << operation->getErrorMessage();
+    }
+
+    // Get the transaction
+    auto result = operation->getResult();
+    ASSERT_TRUE(result.isObject());
+    auto hexTx = result["hex"].getValStr();
+    CDataStream ss(ParseHex(hexTx), SER_NETWORK, PROTOCOL_VERSION);
+    CTransaction tx;
+    ss >> tx;
+    ASSERT_FALSE(tx.vShieldedOutput.empty());
+
+    // We shouldn't be able to decrypt with the empty ovk
+    EXPECT_FALSE(AttemptSaplingOutDecryption(
+        tx.vShieldedOutput[0].outCiphertext,
+        uint256(),
+        tx.vShieldedOutput[0].cv,
+        tx.vShieldedOutput[0].cmu,
+        tx.vShieldedOutput[0].ephemeralKey));
+
+    // We shouldn't be able to decrypt with a random ovk
+    EXPECT_FALSE(AttemptSaplingOutDecryption(
+        tx.vShieldedOutput[0].outCiphertext,
+        random_uint256(),
+        tx.vShieldedOutput[0].cv,
+        tx.vShieldedOutput[0].cmu,
+        tx.vShieldedOutput[0].ephemeralKey));
+
+    auto accountKey = pwalletMain->GetLegacyAccountKey().ToAccountPubKey();
+    auto ovks = accountKey.GetOVKsForShielding();
+    // We should not be able to decrypt with the internal change OVK for shielding
+    EXPECT_FALSE(AttemptSaplingOutDecryption(
+        tx.vShieldedOutput[0].outCiphertext,
+        ovks.first,
+        tx.vShieldedOutput[0].cv,
+        tx.vShieldedOutput[0].cmu,
+        tx.vShieldedOutput[0].ephemeralKey));
+    // We should be able to decrypt with the external OVK for shielding
+    EXPECT_TRUE(AttemptSaplingOutDecryption(
+        tx.vShieldedOutput[0].outCiphertext,
+        ovks.second,
+        tx.vShieldedOutput[0].cv,
+        tx.vShieldedOutput[0].cmu,
+        tx.vShieldedOutput[0].ephemeralKey));
+
+    // Tear down
+    chainActive.SetTip(NULL);
+    mapBlockIndex.erase(blockHash);
+    mapArgs.erase("-developersapling");
+    mapArgs.erase("-experimentalfeatures");
```

Comment:
I think `-developersapling` no longer exists.

Reply:
Also, please file an issue to add Orchard to these tests.

Reply:
it indeed doesn't exist anymore

Reply:
issue created https://github.com/zcash/zcash/issues/5911